server/redis.py

```diff
@@ -8,9 +9,22 @@
     notify_feedback,
 )
 
+from typing import Dict
+
+
 r = redis.Redis(host=REDIS_HOST, port=REDIS_LOCAL_PORT, db=0, charset="utf-8", decode_responses=True)
 
 
+def append_to_stream(key: str, data: Dict):
+    try:
+        parsed_data = {k: json.dumps(v) if type(v) == dict else v for k, v in data.items()}
```

Comment:
NIT: Maybe it would be cleaner to use a custom JSON Encoder in this line. https://alexis-gomes19.medium.com/custom-json-encoder-with-python-f52c91b48cd2